concordium-contracts-common/src/types.rs

```diff
@@ -1289,6 +1405,20 @@ impl AttributeValue {
     pub fn is_empty(&self) -> bool { self.len() == 0 }
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for AttributeValue {
+    fn arbitrary(g: &mut Gen) -> AttributeValue {
+        AttributeValue {
+            inner: [0u8; 32].map(|_| quickcheck::Arbitrary::arbitrary(g)),
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        // todo: implement a shinker if it makes sense
```

Comment:
The first byte encodes the length of the attribute, so you shrink the length and either set the remaining bytes to 0 or leave it as is.

Reply:
@Bargsteen If the first byte is the length, should then `arbitrary`  take this into account? Could you elaborate on how it works exactly? Does the first byte determine how many remaining bytes (out of 32) to consider?

Reply:
I mean, that is up to us, but that was what I suggested. Yes, the length describes how many of the remaining 31 bytes contain actual data.

Reply:
The encoding is pretty precise. Attribute values are never longer than 31 bytes.

And it matters for the meaning what the length is. So please generate
- first byte is a value between 0 and 31, inclusive
- followed by this many bytes of arbitrary data.

Reply:
Done.